tools/clang/test/DXILValidation/clusterid_passing.ll

```diff
@@ -0,0 +1,74 @@
+; RUN: %dxv %s | FileCheck %s
+
+; CHECK: Validation succeeded.
+
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+%dx.types.Handle = type { i8* }
+%struct.Payload = type { <4 x float> }
+%struct.BuiltInTriangleIntersectionAttributes = type { <2 x float> }
+%dx.types.ResourceProperties = type { i32, i32 }
+%dx.types.HitObject = type { i8* }
+
+; Function Attrs: nounwind
+define void @"\01?test_closesthit@@YAXUPayload@@UBuiltInTriangleIntersectionAttributes@@@Z"(%struct.Payload* noalias nocapture %payload, %struct.BuiltInTriangleIntersectionAttributes* nocapture readnone %attr) #0 {
+  %1 = call i32 @dx.op.clusterID(i32 2147483651)  ; ClusterID()
```

Comment:
When this is round-tripped through llvm, `i32 2147483651` will become `i32 -2147483645`. In lowering tests, for example, CHECK lines will need to use `i32 -2147483645` instead of the unsigned value.

Using the unsigned value here could create confusion and additional difficulty for updating tests when moving ops to final DXIL because one will need to use two search/replace expressions to update each opcode, in case the unsigned form is used somewhere.

For this reason, I recommend considering the signed i32 form the canonical one to use in all cases for consistency.

Reply:
```suggestion
  %1 = call i32 @dx.op.clusterID(i32 -2147483645)  ; ClusterID()
```